### PR TITLE
Added node to node palette and allowed drag on to graph

### PR DESCRIPTION
**Pull request by mddvisu**
Adds a node to the node palette and allows the user to put it on the graph. Also has two data pins, one for input one for output. The node is currently called "Small Operator" under the Math section. I did this without creating any more inherited classes of Node (Small Operator is an instance of just Node). This is just a draft.

Progress

### CODE REVIEW

File: Gems/ScriptCanvas/Code/Editor/View/Widgets/NodePalette/NodePaletteModel.cpp [hunk 1]
```diff
@@ -783,6 +796,10 @@ namespace
 
         // Populates the NodePalette with Properties reflected directly on the BehaviorContext
         PopulateBehaviorContextGlobalProperties(nodePaletteModel, *behaviorContext);
+
+        // Populates the NodePalette with nodes that are instances of the Node class instead of inherited classes of Node
+        PopulateConfiguredNodeInformationRatherThanMoreClassHierarchy(nodePaletteModel);
```
Comment (carlitosan) [inline Gems/ScriptCanvas/Code/Editor/View/Widgets/NodePalette/NodePaletteModel.cpp:801]: LOL. Change name to `PopulateWithDataDrivenNodes` or something like that.

File: Gems/ScriptCanvas/Code/Editor/Nodes/NodeCreateUtils.cpp [hunk 2]
```diff
@@ -106,7 +106,17 @@ namespace ScriptCanvasEditor::Nodes
         AZ::Entity* scriptCanvasEntity{ aznew AZ::Entity };
         scriptCanvasEntity->Init();
         nodeIdPair.m_scriptCanvasId = scriptCanvasEntity->GetId();
-        ScriptCanvas::SystemRequestBus::BroadcastResult(node, &ScriptCanvas::SystemRequests::CreateNodeOnEntity, scriptCanvasEntity->GetId(), scriptCanvasId, classId);
+        if (classId == "")
```
Comment (carlitosan) [inline Gems/ScriptCanvas/Code/Editor/Nodes/NodeCreateUtils.cpp:109]: `if (classId.IsNull())`

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/ScriptCanvasBus.h [hunk 3]
```diff
@@ -74,6 +74,8 @@ namespace ScriptCanvas
         //! Given the ClassData for a type create a Script Canvas Node Component on the supplied entity
         virtual Node* CreateNodeOnEntity(const AZ::EntityId& entityId, ScriptCanvasId scriptCanvasId, const AZ::Uuid& nodeType) = 0;
 
+        virtual Node* CreateNodeOnEntityWithoutUuid(const AZ::EntityId& entityId, ScriptCanvasId scriptCanvasId) = 0;
```
Comment (carlitosan) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/ScriptCanvasBus.h:77]: Cut this, and make conditional logic on CreateNodeOnEntity

**Review by carlitosan — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Editor/View/Widgets/NodePalette/NodePaletteModel.cpp [hunk 4]
```diff
@@ -741,6 +741,19 @@ namespace
         }
     }
 
+    void PopulateDataDrivenNodes(
+        ScriptCanvasEditor::NodePaletteModel& nodePaletteModel)
+    {
+
+        // Create the small operator node information
+        ScriptCanvasEditor::RegisterNodeInformation nodeInformation;
```
Comment (onecent1101) [inline Gems/ScriptCanvas/Code/Editor/View/Widgets/NodePalette/NodePaletteModel.cpp:749]: Make sure we have task to clean this temporary code

File: Gems/ScriptCanvas/Code/Source/SystemComponent.cpp [hunk 5]
```diff
@@ -226,22 +226,61 @@ namespace ScriptCanvas
...
+                return node;
+            }
+        }
+        else
         {
             AZ::Entity* nodeEntity = nullptr;
             AZ::ComponentApplicationBus::BroadcastResult(nodeEntity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
-            ScriptCanvas::Node* node = reinterpret_cast<ScriptCanvas::Node*>(classData->m_factory->Create(classData->m_name));
-            AZ_Assert(node, "ClassData (%s) does not correspond to a supported ScriptCanvas Node", classData->m_name);
+            ScriptCanvas::Node* node = aznew Node;
```
Comment (onecent1101) [inline Gems/ScriptCanvas/Code/Source/SystemComponent.cpp:258]: Make sure we have task to clean this temporary code

**Review by onecent1101 — APPROVED** [review]
It is a good starting point, I think you should change your PR ready to review and merge it

**Pull request merged**